lib/resolve/npm.js

```diff
@@ -39,6 +48,66 @@ module.exports = function resolveNpm (pkg) {
     }
     throw err
   }
+
+  function getAllVersionsAndMatchOnClient(pkg) {
+    return Promise.resolve()
+      .then(_ => url.resolve(registryUrl(pkg.scope), pkg.name))
+      .then(url => got(url))
+      .then(res => JSON.parse(res.body))
+      .then(res => {
+        res = pickVersionFromRegistryDocument(res, pkg)
+        return {
+          fullname: '' + res.name.replace('/', '!') + '@' + res.version,
+          version: res.version, // used for displaying
+          dist: res.dist
+        }
+      })
+      .catch(errify)
+  }
+}
+
+function pickVersionFromRegistryDocument(pkg, dep) {
+  // the majority of this implementation was copied from npm:
+  // https://github.com/npm/npm/blob/master/lib/fetch-package-metadata.js#L122-L163
+
+  var versions = Object.keys(pkg.versions).sort(semver.rcompare)
+
+  if (dep.type === 'tag') {
```

Comment:
there's also type `version` (eg, `lodash@4.0.0`) and `range` (eg, `lodash@^4`) by the way.  this is looking good!